platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/RootComponentResolutionIntegrationTest.groovy

```diff
@@ -128,13 +128,14 @@ class RootComponentResolutionIntegrationTest extends AbstractIntegrationSpec {
             }
         """
 
-        executer.expectDocumentedDeprecationWarning("While resolving configuration 'conf', it was also selected as a variant. Configurations should not act as both a resolution root and a variant simultaneously. Depending on the resolved configuration in this manner has been deprecated. This will fail with an error in Gradle 9.0. Be sure to mark configurations meant for resolution as canBeConsumed=false or use the 'resolvable(String)' configuration factory method to create them. Consult the upgrading guide for further information: https://docs.gradle.org/current/userguide/upgrading_version_8.html#depending_on_root_configuration")
+        when:
+        fails("resolve")
 
-        expect:
-        succeeds("resolve")
+        then:
+        failure.assertHasCause("Cannot select root node 'conf' as a variant. Configurations should not act as both a resolution root and a variant simultaneously. Be sure to mark configurations meant for resolution as canBeConsumed=false or use the 'resolvable(String)' configuration factory method to create them.")
     }
 
-    def "configuration can resolve itself and reselect artifacts"() {
+    def "configuration cannot resolve itself and reselect artifacts"() {
```

Comment:
❓ Is this really a different case from the one above?  Doesn't the resolution happen first, so the reselection is irrelevant as that code never even runs due to the earlier failure?

Reply:
Yeah the error happens at graph resolution. Removed this test.